### PR TITLE
Add command to print all debugger help to html file

### PR DESCRIPTION
**Pull request by vadosnaprimer**
Also tweak static_help_list spacing to always pad commands and never pad explanations.

Implementation improvements compared to https://github.com/mamedev/mame/pull/2926:
- don't leak open file
- drop globals and macros
- use `std::ofstream` and `util::stream_format`
- simplify error handling
- make command name more meaningful and provide better explanation of it
- add optional filename for the command

### CODE REVIEW

**Comment by vadosnaprimer** [comment]
@cuavas Check please.

The only part I didn't explicitly do is checking if the file already exists, because no debugger command that writes to file does it. If it fails to open the file, it will report an error, this is what all such commands do instead.

Here's the current output:
https://vadosnaprimer.github.io/debughelp.html

**Comment by Firehawke** [comment]
I'd like to start with the following questions:

1. Why do we need and want to be *generating* and *outputting* HTML from inside MAME?
2. Why does it need to be dynamically generated in the first place?
3. Why can't this be part of the documentation that's already being generated into PDF *and* HTML format?

In short, why do we need to be doing it this way-- why is this better than writing proper standardized documentation?

**Comment by vadosnaprimer** [comment]
The main goal is to provide a document that has links to all the available commands (which I also ordered alphabetically). That way one could read whatever is needed using 1 mouse click, instead of typing commands manually. Other downsides of the debugger console output are: it can't be copied, it's spaced poorly, the window can't be made arbitrarily high (because it takes away the size of the debugger.

Dynamic generation guarantees that the doc is always up-to-date with the latest version of debugger help. And it's always easier to only update one thing than 2, especially if they aren't supposed to be identical.

It _can_ be part of standard documentation, I always said that generating a .rts file in a similar manner is my next plan. But right now there's about 80 commands, putting them all into the navigation panel on the left (for quick linking and navigation) feels like an overkill, even though they can have anchors with no harm. Another problem is, docs are only available in the compiled from on the Internet, and when one is debugging something, they should not fully depend on having Internet connection. Local file is supposed to be a nicely formatted and efficient to use document that's always instantly available.

In general, whenever someone is debugging or hacking, there's little need in the exhaustive book of all the docs every time, just the dedicated page of info is usually enough. The page I make is easy to access and browse. And for the sake of completeness, the traditional doc file will also be generated.

**Comment by Firehawke** [comment]
Docs can be compiled to PDF, HTML or a dozen other forms right now right from the MAME source code:

"Please use \`make <target>' where <target> is one of"
"  html       to make standalone HTML files"
"  dirhtml    to make HTML files named index.html in directories"
"  singlehtml to make a single large HTML file"
"  pickle     to make pickle files"
"  json       to make JSON files"
"  htmlhelp   to make HTML files and a HTML help project"
"  qthelp     to make HTML files and a qthelp project"
"  devhelp    to make HTML files and a Devhelp project"
"  epub       to make an epub"
"  latex      to make LaTeX files, you can set PAPER=a4 or PAPER=letter"
"  latexpdf   to make LaTeX files and run them through pdflatex"
"  text       to make text files"
"  man        to make manual pages"
"  texinfo    to make Texinfo files"
"  info       to make Texinfo files and run them through makeinfo"
"  gettext    to make PO message catalogs"
"  changes    to make an overview of all changed/added/deprecated items"
"  linkcheck  to check all external links for integrity"
"  doctest    to run all doctests embedded in the documentation (if enabled)"

Put simply, I do not believe that you're approaching this the correct way. Debugger commands should not be changing frequently, and having user-facing documentation in ONE set of maintained documents-- instead of two places-- is the reason we started the documentation redesign a few years ago. 

Frankly, I'd really rather take your design, port it up to the RTD formatting, and maintain it that way. If you wanted to view it as a local HTML, you can. If you wanted to view it as a local PDF, you can. Want it as a man file? ePub? LaTeX? All viable from the package.

You haven't shown me a single benefit from this that justifies why the additional pain of having to maintain the debugger docs through a bolted-on set of horrors.

**Comment by rb6502** [comment]
I didn't know we had that infrastructure for documentation or I would've cut this off long ago.  Let's get it ported to proper documentation and put a copy on the website.

**Comment by cracyc** [comment]
This goes for https://github.com/mamedev/mame/issues/3000 too.

**Comment by vadosnaprimer** [comment]
A bit of investigation proved that most of this PRs goals can be satisfied by using the existing docs: having lightweight local docs package, having anchor links withing one page. I'm going to use this approach to generate an rst file instead.

**Comment by stilett0** [comment]
@Firehawke finally nailed down the bit that was bothering me about this: at some point, we stopped delivering offline docs beyond the Linux port's man pages, which is what makes this patch somewhat interesting. Back in the day, MAME provided, with their binary Windows downloads, a full set of docs. I think I'm going to look into when, and why - I have a feeling it wasn't fully intentional, and probably happened sometime around a release packager change in personnel.
